%YAML:1.0

# Bag Info
preRectified: 1
startTime: 8
bagPath: "/media/ygling2008/DATA/KITTI/stereo/21/"
lidarPath: "/media/ygling2008/DATA/KITTI/lidar/21/velodyne/"
gtPoseFile: "/media/ygling2008/DATA/KITTI/poses/21.txt"
imageTotalNum: 2720
flipStereo: 0

# Tracker
equalized: 0
MIN_DIST: 20
F_THRESHOLD: 3.0
SHOW_TRACK: 1
MIN_PARALLAX: 10.0
FREQ: 10
skipFrames: 0
TrackerTotalTime: 100.0

# Loop
loop_MAX_FEATURES: 300

# SGBM
useCUDA: 1
upMargin: 50
# upMargin: 50
downMargin: 60
# downMargin: 60
leftMargin: 128
# leftMargin: 128
rightMargin: 20
# rightMargin: 20
maxDisparity: 64
SADWindowSize: 11

# Debug Info
displayViewer: 1

# Mesh parameters
updateMeshFlag: 0
cubicBound: 30.0
updateNonKF: 0
cvWaitTime: 1
renderingK: 100
useColor: 1
useCarving: 1
nearPlaneDist: 0.1
farPlaneDist: 20.0
chunkSizeX: 8
chunkSizeY: 8
chunkSizeZ: 8
voxelResolution: 0.2
weight: 1
carvingDist: 0.05
truncationDistConst: 0.001504
truncationDistLinear: 0.00152
truncationDistQuad: 0.0019
truncationDistScale: 8.0
downSampleTimes: 1
updateMeshInterval: 0
delayLoopTime: 0

#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------

# Camera calibration and distortion parameters (OpenCV) 
Camera.fx: 718.856
Camera.fy: 718.856
Camera.cx: 607.1928
Camera.cy: 185.2157

Camera.k1: 0.0
Camera.k2: 0.0
Camera.p1: 0.0
Camera.p2: 0.0

# Camera frames per second 
Camera.fps: 5.0

# stereo baseline times fx
Camera.bf: 386.1448

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 1

# Close/Far threshold. Baseline times.
ThDepth: 40

#--------------------------------------------------------------------------------------------
# ORB Parameters
#--------------------------------------------------------------------------------------------

# ORB Extractor: Number of features per image
ORBextractor.nFeatures: 2000

# ORB Extractor: Scale factor between levels in the scale pyramid 	
ORBextractor.scaleFactor: 1.2

# ORB Extractor: Number of levels in the scale pyramid	
ORBextractor.nLevels: 8

# ORB Extractor: Fast threshold
# Image is divided in a grid. At each cell FAST are extracted imposing a minimum response.
# Firstly we impose iniThFAST. If no corners are detected we impose a lower value minThFAST
# You can lower these values if your images have low contrast			
ORBextractor.iniThFAST: 20
ORBextractor.minThFAST: 7

#--------------------------------------------------------------------------------------------
# Viewer Parameters
#--------------------------------------------------------------------------------------------
Viewer.KeyFrameSize: 2
Viewer.KeyFrameLineWidth: 1
Viewer.GraphLineWidth: 0.9
Viewer.MeshLineWidth: 0.8
Viewer.PointSize: 2
Viewer.CameraSize: 3
Viewer.CameraLineWidth: 3
#Viewer.ViewpointX: 0
#Viewer.ViewpointY: -100
#Viewer.ViewpointZ: -5
Viewer.ViewpointX: 0
Viewer.ViewpointY: 0
Viewer.ViewpointZ: -10
Viewer.ViewpointF: 707.0912
Viewer.TopDownZ: -500

